### PR TITLE
RN-188 Enabling and disabling of selected options

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
RN-188

### Change
1. Store a new config `isDisable` (name is negotiable) in each `aggregate` and `transform` object, which indicates if associated step is selected to fetch preview data.
2. Select all button.

### Screenshots:
https://linear.app/bes/issue/RN-188#comment-3dedc555

### CODE REVIEW

File: packages/admin-panel/src/VizBuilderApp/context/VizConfig.js [hunk 1]
```diff
@@ -116,24 +116,57 @@ const useConfigStore = () => {
...
 
   return (
-    <Provider value={store} displayName="VizBuilder">
-      {children}
-    </Provider>
+    <VisualisationContext.Provider
+      value={{
+        visualisationForFetchingData: removeIsDisableConfig(filterDisableSteps(visualisation)),
+        visualisation: removeIsDisableConfig(visualisation),
+      }}
```
Comment (billli0) [inline packages/admin-panel/src/VizBuilderApp/context/VizConfig.js:153]: Provide a new context `VisualisationContext`, which works like a selector but no caching ability.

**Review by rohan-bes — CHANGES_REQUESTED** [review]
Looks great @billli0 this is such awesome functionality!

Couple of minor renames, and a bit of a refactor request, lemme know what you think 👍

File: packages/admin-panel/src/VizBuilderApp/components/DataLibrary/AggregationDataLibrary.js [hunk 2]
```diff
@@ -46,14 +69,36 @@ export const AggregationDataLibrary = ({ aggregate, onAggregateChange, onInvalid
...
+            newSelectedAggregations[index].isDisable = newValue.isDisable;
             onAggregateChange(valueToAggregate(newSelectedAggregations));
           }}
           onRemove={onRemove}
           setEdittingOption={setEdittingOption}
           onInvalidChange={onInvalidChange}
         />
       )}
+      header={
+        <ColHeader>
```
[the remark this answers is not in the document]
Comment (billli0) [reply packages/admin-panel/src/VizBuilderApp/components/DataLibrary/AggregationDataLibrary.js:83]: My original thought is to make thing clearer than passing a prop. One of the benefits is we can avoid to pass `onChangeForCheckbox` or `checkboxConfig` for all `setState()` hooks, which makes code more readable and straigtfoward. 

It seems to bring more confusion than readability. I'm happy to pass prop `supportsDisableAll` to enable the checkbox.

File: packages/admin-panel/src/VizBuilderApp/components/DataLibrary/AggregationDataLibrary.js [hunk 3]
```diff
@@ -46,14 +69,36 @@ export const AggregationDataLibrary = ({ aggregate, onAggregateChange, onInvalid
...
+          <Checkbox
+            checkedIcon={<LibraryAddCheckOutlinedIcon />}
+            checked={!isDisableAll}
+            onChange={() => {
+              const newSelectedAggregations = Array.from(value).map(v => {
+                const newV = { ...v };
+                newV.isDisable = !isDisableAll;
+                return newV;
+              });
```
[the remark this answers is not in the document]
Comment (billli0) [reply packages/admin-panel/src/VizBuilderApp/components/DataLibrary/AggregationDataLibrary.js:92]: Yeah, will change it to `baseValue` to make more sense.

**Review by rohan-bes — APPROVED** [review]
Nice work @billli0 super useful!

**Pull request merged**